### ISSUE TITLE
Image Field: wrong preview size before initial save

### ISSUE THREAD

**Issue by mjot** [AdvancedCustomFields/acf#505]
To demonstrate the problem, a single image field is needed and a custom image size. I'll just use a portrait format with the dimensions 300x420 here for easier demonstration. The image size is registered using this function in the function.php:

`add_image_size('assets', 300, 420, true);`

Then one image is uploaded, and all the needed sizes are generated without a problem:

```
-rw-r--r-- 1 wp wp 1084 May 20 15:42 test-600x600-1-150x150.png
-rw-r--r-- 1 wp wp 2704 May 20 15:42 test-600x600-1-300x300.png
-rw-r--r-- 1 wp wp 4662 May 20 15:42 test-600x600-1-300x420.png
-rw-r--r-- 1 wp wp 1840 May 20 15:42 test-600x600-1.png
```

Here you can see the field config:
![field-config](https://user-images.githubusercontent.com/1445991/118990212-f909fb00-b982-11eb-9f55-1c1a3424ccfd.png)

I then open a post and select the image using the Media Library. Without saving the post, the image is displayed inside the field with the default format "Medium", but not with my configured size "Assets":

![before-save](https://user-images.githubusercontent.com/1445991/118990514-3f5f5a00-b983-11eb-8234-dacb4532f16c.png)

If I now click on save, the page reloads and the Image Field shows the mage in the correct size:
![after-save](https://user-images.githubusercontent.com/1445991/118990650-5bfb9200-b983-11eb-8627-68e1c3b72f2f.png)

Would it be possible to correct this so, that the correct preview size is displayed directly after selecting the image?
I have already discussed the problem with your support and it was recommended to me to make an entry here. :)

Thanks,
Martin

**Comment by elliotcondon** [comment]
Hi @mjot 

Thank you for bringing this issue to our attention with such a detailed bug report 👍.

I'm glad to share with you that I successfully replicated and fixed the issue. This fix will be included within the next release. 
For early access to the patch, please find a link to a Gist containing updated `media.php` code to replace the one found within the plugin file `includes/media.php`:
https://gist.github.com/elliotcondon/06c5be1d7425135553bb13c8c9734a6f

**Comment by mjot** [comment]
Thanks @elliotcondon,

i tested it with ACF Pro 5.9.6 and it works like a charm. :)

**Comment by mjot** [comment]
Hello @elliotcondon,

I found a small error in the debug log after all:

`PHP Notice:  Undefined variable: field_key in advanced-custom-fields-pro/includes/media.php on line 148`

Here is Line 148 - 150
```
if( $field_key ) {
	$field = acf_get_field( $field_key );
}
```

I think this should fix it?
```
if( isset($field_key) ) {
	$field = acf_get_field( $field_key );
}
```

Thanks for your second look at this matter :)

**Comment by elliotcondon** [comment]
@mjot Agreed. Using `isset()` here is a neat solution. I'll update the Gist. 👍

**Comment by mjot** [comment]
Thanks again @elliotcondon. :)
